### PR TITLE
Drain node cmd

### PR DESCRIPTION
**Pull request by ffilippopoulos**
Add a command to drain nodes.

Reshapes replica deletion functions to be reused by the new commnad

### CODE REVIEW

File: cmd/delete_replica.go [hunk 1]
```diff
@@ -6,16 +6,10 @@ import (
...
 
 	"github.com/spf13/cobra"
-
-	lh_client "github.com/longhorn/longhorn-manager/client"
-)
-
-var (
-	ReplicaName   string           // Name of replica to be deleted
-	ReplicaVolume lh_client.Volume // Volume the replica belongs to
+	//lh_client "github.com/longhorn/longhorn-manager/client"
```
Comment (george-angel) [inline cmd/delete_replica.go]: rm

File: cmd/drain_node.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+							))
+							return
+						}
+						fmt.Println(fmt.Sprintf(
+							"error %v deleting replica: %s, retrying",
+							err,
+							replicaName,
+						))
+						// sleep before retrying
+						time.Sleep(time.Duration(1 * time.Second))
```
Comment (george-angel) [inline cmd/drain_node.go:133]: Perhaps a timeout here?

Comment (ffilippopoulos) [reply cmd/drain_node.go:133]: timeout is a good idea. I also think that it should be configurable via a flag on the command

**Comment by ffilippopoulos** [comment]
Closing, let's start with something simpler that just exits on failure. If that gets too difficult to handle from a script we can revisit the timeout and retry.

Closing in favour of https://github.com/utilitywarehouse/lhctl/pull/12

**Pull request closed without merging**